### ISSUE TITLE
ArrayIndexOutofBoundsException when using tint() or VLW fonts in 3.3.1

### ISSUE THREAD

**Issue by ewhynott** [processing/processing#5028]
I'm new to Processing, but I only started having issues when I downloaded the latest version. When I try to load any font, it only works with specific text within the text function. For instance, it might display "test" but "tent" will not work. I get an error message "Target VM failed to initialize" as well as " ArrayIndexOutofBoundsException18". When I run the sketch, it crashes and displays the message: "Java (TM) Platform SE binary is not Responding".
If it helps, I have Windows 10 and the latest version of Processing.
Thank you in advance, any help is appreciated!

```processing
//my code:
PFont myFont;

void setup(){
size(800,800);
 myFont = loadFont("LithosPro-Black-30.vlw");//the name is copied from sketch folder
}

void draw(){
  
  background(0);//keep in draw to make text move around without paint trail
 textFont(myFont); //set "family font" --this line--before changing scale
    fill(255,0,0);//color of text
  textSize(20);//mouseX makes it change scale
 
  text("poetry",400,400);//"text" is what is being written
 
}
```

**Comment by benfry** [comment]
Is there a file named `LithosPro-Black-30.vlw` in the `data` folder of this sketch?

**Comment by ewhynott** [comment]
Thanks for the fast reply! There is. I have had the issue with multiple fonts, all in the sketch folder. My professor was unable to determine the cause of the issue, he said the code was sound but for some reason would not execute on my laptop. I'd be happy to send screen shots of the specific error messages if it would help?

**Comment by waikman1** [comment]
I've had a similar error this morning in 3.3.1 when resizing and tinting mulitple png's. The sketch was running perfectly in earlier processing versions. I've tested in processing 2.2.1 and its solved the problem.

**Comment by benfry** [comment]
Yeah, I've been able to reproduce the problem. Sorry about that.

@JakubValtar `tint()` is broken for images (breaking VLW fonts as well)… suspect this is from the pixel changes in `PGraphicsJava2D`. Do you have a minute to take a look?

**Comment by JakubValtar** [comment]
@benfry I will fix it today.

**Comment by benfry** [comment]
Great, thanks. I was able to reproduce with just the `LoadDisplayImage` example and adding a `tint()` line. (As well as using any `loadFont()` and trying to draw text.)

**Comment by JakubValtar** [comment]
@benfry Done. My bad. I went quickly through the whole diff and also fixed a similar typo in FX2D.

**Comment by ewhynott** [comment]
Thank you so much for your help! I am still having the ArrayIndexOutofBoundsException18 issue, is there a new version of Processing or something I should do to allow for the corrections made by @JakubValtar? Thank you for your patience, again I only just started learning the language!

**Comment by JakubValtar** [comment]
@ewhynott Please use 3.3 for now. We need to merge the fix and release a new version of Processing. You will be able to download the fixed version as soon as Processing 3.3.2 appears on the download page. https://processing.org/download/

**Comment by ewhynott** [comment]
Awesome, I can't thank you enough for your help! Thank you for being so available and patient! Fingers crossed this is the last time I have to reach out :)
